src/main/java/spoon/support/compiler/jdt/JDTTreeBuilder.java

```diff
@@ -935,7 +936,11 @@ public boolean visit(BreakStatement breakStatement, BlockScope scope) {
 
 	@Override
 	public boolean visit(CastExpression castExpression, BlockScope scope) {
-		context.casts.add(this.references.buildTypeReference(castExpression.type, scope));
+		CastInfo ci = new CastInfo();
+		//the 8 bits from 21 to 28 represents number of enclosing brackets
```

Comment:
Just for my information, because I'm curious ;) It's something you obtained by reverse engineering or there's a documentation somewhere with that info?

Reply:
There is a documentation of these bits in source code of compiler ... and I needed some luck to found it ;-)